### ISSUE TITLE
GoDot 4 - doesn't work

### ISSUE THREAD

**Issue by mattkanwisher** [Greaby/godot-qrcode-generator#3]
Your project doesn't seem to work under GoDot 4. I'll try and figure out and open a PR

**Comment by Greaby** [comment]
Thanks, I will make a branch for Godot 4.

**Comment by Greaby** [comment]
Created the [branch](/Greaby/godot-qrcode-generator/tree/godot-4) for Godot 4.
The QR code is generated but seems corrupted.

**Comment by mattkanwisher** [comment]
you saying its corrupted from my PR? or I need to point my PR at your new branch

**Comment by mattkanwisher** [comment]
Yeah I can reproduce the corrupted QR code, it seems to be related to the changes in Main. I ported mine a month ago and the godot 4 upgrade worked fine. I'm trying to see what broke in the error correction code

**Comment by RAbsurdum** [comment]
One issue I've discovered is that references to the **const** arrays are no longer treated the same; they cannot be "reversed()" in Godot 4 the way they were able to be "inverted()" in Godot 3. Using "duplicate()" when adding or appending those **const** arrays brought the Godot 4 output image much closer to that produced by the Godot 3 code; however, there are still discrepancies. Curiously, these have tended to be localized in the output image. For example, this image -- https://i.imgur.com/TGOrVjE.png -- was produced by the Godot 3 code (using input text of "https://www.google.com" and using ERROR_CORRECT_LEVEL.LOW); the same input text and ERROR_CORRECT_LEVEL using Godot 4 created this image -- https://i.imgur.com/Uy5uwtM.png. Note that all of the discrepancies/differences are located within the left half of the image. No idea if this is helpful information, but I am also hoping for a working version of this code for Godot 4.

Edit: Ok, I finally hunted down the issue. In addition to adding ".duplicate()" to each of the **const** arrays referenced in the method "_set_info_segments()", in the method "_split_data_into_blocks()" just below it is necessary to change the following line:

`var block: Array = qr_data_list.slice(off, end-1)`
to
`var block: Array = qr_data_list.slice(off, end)`

Basically, in Godot 3 the slice function is inclusive both at the beginning and end of the slice, whereas in Godot 4+ the slice function is only inclusive at the beginning. With these changes, the code works perfectly in Godot 4.

Thank you so much for making this project available to the community!
